metrics_collector/volume_collector.go

```diff
@@ -214,6 +230,31 @@ func (vc *VolumeCollector) collectMetrics(ch chan<- prometheus.Metric, v *longho
 	ch <- prometheus.MustNewConstMetric(vc.volumePerfMetrics.iopsMetrics.write.Desc, vc.volumePerfMetrics.iopsMetrics.write.Type, float64(vc.getVolumeWriteIOPS(metrics)), vc.currentNodeID, v.Name, v.Status.KubernetesStatus.PVCName, v.Status.KubernetesStatus.Namespace)
 	ch <- prometheus.MustNewConstMetric(vc.volumePerfMetrics.latencyMetrics.read.Desc, vc.volumePerfMetrics.latencyMetrics.read.Type, float64(vc.getVolumeReadLatency(metrics)), vc.currentNodeID, v.Name, v.Status.KubernetesStatus.PVCName, v.Status.KubernetesStatus.Namespace)
 	ch <- prometheus.MustNewConstMetric(vc.volumePerfMetrics.latencyMetrics.write.Desc, vc.volumePerfMetrics.latencyMetrics.write.Type, float64(vc.getVolumeWriteLatency(metrics)), vc.currentNodeID, v.Name, v.Status.KubernetesStatus.PVCName, v.Status.KubernetesStatus.Namespace)
+
+	fileSystemReadOnlyCondition := types.GetCondition(e.Status.Conditions, imtypes.EngineConditionFilesystemReadOnly)
+	isPVMountOptionReadOnly := false
+	kubeStatus := v.Status.KubernetesStatus
+	if kubeStatus.PVName != "" {
+		pv, err := vc.ds.GetPersistentVolumeRO(kubeStatus.PVName)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				vc.logger.WithError(err).Warnf("Failed to get PV when checking the mount option of the volume")
+				return
+			}
+		}
+		if pv != nil {
+			for _, opt := range pv.Spec.MountOptions {
+				if opt == "ro" {
+					isPVMountOptionReadOnly = true
+					break
+				}
+			}
+		}
```

Comment:
It looks like there is a duplicated logic in the volume controller. We could consider abstracting this into a reusable function. What do you think?

Reply:
Hi @c3y1huang 
Fixed.
For both case, I update the implementation to return error if `GetPersistentVolumeRO()` fails to retrieve PV. (`kubeStatus.PVName` is not empty)
And return immediately.
No need to handle error separately.